projects/archiver/index.ts

```diff
@@ -0,0 +1,92 @@
+require('dotenv').config()
+
+import { Handler } from 'aws-lambda'
+import { getIssue, getFront } from './downloader'
+import { attempt, hasFailed, hasSucceeded, Attempt } from '../backend/utils/try'
+import { Front, issuePath, frontPath } from './common'
+import { upload } from './upload'
+import { zip } from './zipper'
+import { getImagesFromFront, uploadImage } from './media'
+import { unnest } from 'ramda'
+import { imageSizes } from '../common/src/index'
+
+const run = async (id: string): Promise<void> => {
+    console.log('Attempting to download issue.')
+    const issue = await attempt(getIssue(id))
+    if (hasFailed(issue)) throw new Error('OH NO')
+    console.log('Downloaded issue', id)
+    const maybeFronts = await Promise.all(
+        issue.fronts.map(
+            async (frontid): Promise<[string, Attempt<Front>]> => [
+                frontid,
+                await attempt(getFront(id, frontid)),
+            ],
+        ),
+    )
+    maybeFronts.forEach(([id, attempt]) => {
+        if (hasFailed(attempt)) {
+            console.warn(`Front ${id} failed with ${attempt.error}`)
+        }
+    })
+
+    console.log('Fetched fronts')
+
+    await Promise.all(
+        maybeFronts.map(async ([frontId, maybeFront]) => {
+            if (hasSucceeded(maybeFront)) {
+                return await attempt(
+                    upload('data', frontPath(id, frontId), maybeFront),
+                )
+            }
+        }),
+    )
+    console.log('Uploaded fronts')
+
+    const images = unnest(
+        maybeFronts
+            .map(([, maybeFront]) => maybeFront)
+            .filter(hasSucceeded)
```

Comment:
Could we do a filter once after fetching so that we don't have do do the check when uploading too? Not a biggy but just reduces the code a touch.